api/lib/routes/scalingHistoryHelper.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+function isMissing(value) {
+  return typeof(value) === "undefined" || value === null;
+}
+
+function isInteger(value) {
+  return Number.isInteger(value);
+}
+exports.validateParameter = function(req) {
+  var appId = req.params.guid;
+  var startTime = req.query["start-time"];
+  var endTime = req.query["end-time"];
+  var order = req.query["order"];
+  var page = req.query["page"];
+  var resultsPerPage = req.query["results-per-page"];
+  if (!appId) {
+    return { valid: false, message: "app guid is required" };
+  }
+
+  if (isMissing(startTime)) {
+    return { valid: false, message: "start-time is required" };
+  }
+  startTime = parseInt(startTime);
+  if (!isInteger(startTime)) {
+    return { valid: false, message: "start-time must be an integer" };
+  }
+
+  if (isMissing(endTime)) {
+    return { valid: false, message: "end-time is required" };
+  }
+  endTime = parseInt(endTime);
+  if (!isInteger(endTime)) {
+    return { valid: false, message: "end-time must be an integer" };
+  }
+
+  if (startTime > endTime) {
+    return { valid: false, message: "start-time must be smaller than end-time" };
+  }
+
+  if (isMissing(page)) {
+    return { valid: false, message: "page is required" };
+  }
+  page = parseInt(page);
+  if (!isInteger(page)) {
+    return { valid: false, message: "page must be an integer" };
+  }
+  if (page < 0) {
+    return { valid: false, message: "page must be greater or equal than 0" };
+  }
+
+  if (isMissing(resultsPerPage)) {
```

Comment:
@kanekoh I meant if the parameters are not provided in the query it should not fail instead use some default values.

> $ cf curl /v2/apps

doesn't fail when page or results-per-page are not provided.